### ISSUE TITLE
Handle symlinks

### ISSUE THREAD

**Issue by altermo** [EmmyLuaLs/emmylua-analyzer-rust#650]
Handle symbolicy linked files and directories as normal ones.

**Bash Script Example**
```bash
mkdir foo
echo "return 0">a.lua
ln -s a.lua foo/a.lua
echo -e "---@type string\nlocal _=require'a'">foo/b.lua
emmylua_check foo
```
should produce same warning as
```bash
mkdir foo
echo "return 0">foo/a.lua
echo -e "---@type string\nlocal _=require'a'">foo/b.lua
emmylua_check foo
```

**Comment by CppCXY** [comment]
After giving it some thought, I’ve decided that this design cannot be changed. If both the original file and its symbolic link exist in the loaded workspace, keeping only one of them would be incorrect, as this would prevent the module system that references it from correctly locating the file. Additionally, in LSP, a file’s identity is described only by its URI, not by its file path, and a URI does not necessarily correspond to an actual file on the local machine. Since emmylua_check and ls share the same system, they cannot be handled specially. If you have such a requirement, I suggest excluding directories through configuration.

**Comment by bycruz** [comment]
Not sure if I understand, is this a hard specific limitation in the architecture for the project? LuaLS handles symlinks (directories) just fine for me.

If symlinks are entirely disallowed, it's a critical blocker for something like https://github.com/lde-org/lde where projects have their dependencies stored in a `target` directory with symlinks inside to specific packages. I don't think symlinks are too exotic of an idea, and they should be quite simple rather than dealing with, say, actually just duplicate files copied.

If possible, I'd like for this project to be compatible with lde, but if it isn't, users will have to be forced onto LuaLS.

**Comment by CppCXY** [comment]
This was my reply from a long time ago and no longer reflects the current reality. I suggest judging based on the current, specific behavior instead.
